apm-agent-plugins/apm-redis-plugin/apm-jedis-plugin/src/main/java/co/elastic/apm/agent/redis/jedis/JedisInstrumentation.java

```diff
@@ -63,9 +70,7 @@ private static void afterSendCommand(@Nullable @Advice.Local("span") Span span,
     @Override
     public ElementMatcher<? super TypeDescription> getTypeMatcher() {
         return named("redis.clients.jedis.Jedis")
-            .or(named("redis.clients.jedis.BinaryJedis"))
-            .or(named("redis.clients.jedis.ShardedJedis"))
-            .or(named("redis.clients.jedis.BinaryShardedJedis"));
+            .or(named("redis.clients.jedis.BinaryJedis"));
```

Comment:
Maybe, instead of excluding the Jedis clients from instrumentation which don't have the `getClient()` method, check `if (thiz instanceof BinaryJedis)` and only then set the destination fields.

Reply:
Of course, if it is required then I will not remove it. I removed it because it seemed redundant (as the sharded client internally calls a non-sharded one- see my question in the top issue comment).

Reply:
Ah, didn't know this but seems you are right 👍